Library/Homebrew/test/test_formula.rb

```diff
@@ -406,3 +428,101 @@ def test_pour_bottle_dsl
     assert f_true.pour_bottle?
   end
 end
+
+class OutdatedVersionsTests < Homebrew::TestCase
+  attr_reader :outdated_prefix, :same_prefix, :greater_prefix, :head_prefix
+  attr_reader :f
+
+  def setup
+    @f = formula { url "foo"; version "1.20" }
+    @outdated_prefix = HOMEBREW_CELLAR/"#{f.name}/1.11"
+    @same_prefix = HOMEBREW_CELLAR/"#{f.name}/1.20"
+    @greater_prefix = HOMEBREW_CELLAR/"#{f.name}/1.21"
+    @head_prefix = HOMEBREW_CELLAR/"#{f.name}/HEAD"
+  end
+
+  def teardown
+    @f.rack.rmtree
+  end
+
+  def setup_tab_for_prefix(prefix, tap_string=nil)
+    prefix.mkpath
+    tab = Tab.empty
+    tab.tabfile = prefix.join("INSTALL_RECEIPT.json")
+    tab.source["tap"] = tap_string if tap_string
+    tab.write
+    tab
+  end
+
+  def test_greater_different_tap_installed
+    setup_tab_for_prefix(greater_prefix, "user/repo")
+    assert_predicate f.outdated_versions, :empty?
+  end
+
+  def test_greater_same_tap_installed
+    f.instance_variable_set(:@tap, CoreTap.instance)
+    setup_tab_for_prefix(greater_prefix, "homebrew/core")
+    assert_predicate f.outdated_versions, :empty?
+  end
+
+  def test_outdated_different_tap_installed
+    outdated_tab = setup_tab_for_prefix(outdated_prefix, "user/repo")
```

Comment:
The return value assigned to the local variable is never used, thus the assignment can be dropped.